Revert shell change

praveenkumar opened the pull request.

praveenkumar committed:
Revert "Refactor shell code in testsuite"

This reverts commit b1002c1e583dc0ca23f1bbb79b1dbcc5eec5e827.

During the use of this commit on crc integration test, we found out
that we do need to depend on single shell instance so easily can able
to replicate the user scenario.

A user usually continue in single shell session, it can be exported some
variable or creating other variables and using those in next commands.
With the current refactor every shell instance is a seperate one for execution
of a command.

praveenkumar committed:
Add test for shell variable

This test make sure the same shell session run through out the feature/scenario.

cfergeau commented on features/testsuite.feature:
```diff
@@ -62,6 +62,24 @@ Quentin check whether their testsuite works properly.
+    """
+    hello
+    """
+
```
`Then stdout should contain "hello"` may be enough? though most other tests are already using `"""`
I'm wondering whether `Scenario: Scenario Variables` isn't already testing something very similar. The test you add seems like it has the same effect as `When setting scenario variable "VAR" to the stdout from executing "echo 'hello'"`

praveenkumar replied:
> seems like it has the same effect as When setting scenario variable "VAR" to the stdout from executing "echo 'hello'"

Effect is same but the underlying implementation is different in case of scenario variable it is using different function.

praveenkumar committed:
In windows the exit code for a non-found command is 0

```
PS > foobar
foobar : The term 'foobar' is not recognized as the name of a cmdlet, function, script file, or operable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:1
+ foobar
+ ~~~~~~
    + CategoryInfo          : ObjectNotFound: (foobar:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

PS > echo $lastexitcode
0
```
